src/treespec/treespec.cpp

```diff
@@ -319,6 +319,44 @@ py::list PyTreeSpec::Entries() const {
     }
 }
 
+py::object PyTreeSpec::Entry(ssize_t index) const {
+    EXPECT_FALSE(m_traversal.empty(), "The tree node traversal is empty.");
+    const Node& root = m_traversal.back();
+    if (index < -root.arity || index >= root.arity) [[unlikely]] {
+        throw py::index_error("PyTreeSpec::Entry() index out of range.");
+    }
+    if (index < 0) [[unlikely]] {
+        index += root.arity;
+    }
+
+    if (root.node_entries) [[unlikely]] {
+        return GET_ITEM_BORROW<py::tuple>(root.node_entries, index);
+    }
+    switch (root.kind) {
+        case PyTreeKind::Tuple:
+        case PyTreeKind::List:
+        case PyTreeKind::NamedTuple:
+        case PyTreeKind::Deque:
+        case PyTreeKind::StructSequence:
+        case PyTreeKind::Custom: {
+            return py::int_(index);
+        }
+
+        case PyTreeKind::Dict:
+        case PyTreeKind::OrderedDict: {
+            return GET_ITEM_BORROW<py::list>(root.node_data, index);
+        }
+        case PyTreeKind::DefaultDict: {
+            return GET_ITEM_BORROW<py::list>(GET_ITEM_BORROW<py::tuple>(root.node_data, 1), index);
+        }
+
+        case PyTreeKind::None:
```

Comment:
Why raise error for None and Leaf?

Reply:
This is unreachable. For `None` and leaf, it will raise `IndexError` due to there are no children.

For `None` and leaf, `root.arity = 0`.

```c++
    if (index < -root.arity || index >= root.arity) [[unlikely]] {
        throw py::index_error("PyTreeSpec::Entry() index out of range.");
    }
```